async-nats/src/jetstream/types.rs

```diff
@@ -0,0 +1,193 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Duration;
+
+use serde::{Deserialize, Serialize};
+use time::serde::rfc3339;
+
+/// `StreamConfig` determines the properties for a stream.
+/// There are sensible defaults for most. If no subjects are
+/// given the name will be used as the only subject.
+#[derive(Debug, Default, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct StreamConfig {
+    /// A name for the Stream. Must not have spaces, tabs or period `.` characters
+    pub name: String,
+    /// How large the Stream may become in total bytes before the configured discard policy kicks in
+    pub max_bytes: i64,
+    /// How large the Stream may become in total messages before the configured discard policy kicks in
+    pub max_msgs: i64,
```

Comment:
Maybe we shouldn't abbreviate message, since the have `Message` as a type it makes sense not to.
```suggestion
    pub max_messages: i64,
```